packages/layout/src/Modal.ts

```diff
@@ -0,0 +1,204 @@
+    }
+
+    closeModal() {
+        if (typeof this._close === "function") {
+            this._close();
+        }
+        this.show(false).render();
+    }
+
+    getRelativeTarget() {
+        let relativeTarget;
+        if (this.relativeTargetId()) {
+            relativeTarget = document.getElementById(this.relativeTargetId());
+            if (relativeTarget) {
+                return relativeTarget;
+            }
+        }
+        if (!relativeTarget) {
+            relativeTarget = this.locateAncestor("layout_Grid");
+            if (relativeTarget && relativeTarget.element) {
+                return relativeTarget.element().node();
+            }
+        }
+        return document.body;
+    }
+
+    setModalSizeLimits() {
+        if (this.minHeight() || this.minWidth()) {
+            this._modal
+                .style("min-height", this.minHeight())
+                .style("min-width", this.minWidth())
+                .style("max-height", this.maxHeight())
+                .style("max-width", this.maxWidth())
+                ;
+        }
+    }
+
+    setFadePosition(rect) {
+        this._fade.node().style.top = rect.top + "px";
+        this._fade.node().style.left = rect.left + "px";
+        this._fade.node().style.width = rect.width + "px";
+        this._fade.node().style.height = rect.height + "px";
+    }
+
+    setModalPosition(rect) {
+        const _node = this._modal.node();
+        const contentRect = _node.getBoundingClientRect();
+        _node.style.position = "fixed";
+        _node.style.top = (rect.top + (rect.height / 2) - (contentRect.height / 2)) + "px";
+        _node.style.left = (rect.left + (rect.width / 2) - (contentRect.width / 2)) + "px";
+        _node.style.width = contentRect.width + "px";
+        _node.style.height = contentRect.height + "px";
+        if (this.minHeight() || this.minWidth()) {
+            _node.style["min-height"] = this.minHeight();
+            _node.style["min-width"] = this.minWidth();
+        }
+    }
+
+    enter(domNode, element) {
+        super.enter(domNode, element);
+        const context = this;
+        this._fade = element.append("div")
+            .classed("layout_Modal-fade", true)
+            .classed("layout_Modal-fadeClickable", this.enableClickFadeToClose())
+            .classed("layout_Modal-fade-hidden", !this.showFade())
+            ;
+        const header_h = this.titleFontSize() * 2;
+        this._modal = element.append("div")
+            .classed("layout_Modal-content", true)
+            ;
+        this._modalHeader = this._modal.append("div")
+            .classed("layout_Modal-header", true)
+            .style("color", this.titleFontColor())
+            .style("font-size", this.titleFontSize() + "px")
+            .style("height", header_h + "px")
+            ;
+        this._modalBody = this._modal.append("div")
+            .classed("layout_Modal-body", true)
+            .style("height", `calc( 100% - ${header_h}px )`)
```

Comment:
I just checked MDN and don't see any reason not to use `calc` - which will potentially help simplify layout code in the future!